RecoLocalCalo/HGCalRecProducers/plugins/HGCalCLUEAlgo.cc

```diff
@@ -216,33 +220,104 @@ math::XYZPoint HGCalCLUEAlgo::calculatePosition(const std::vector<int>& v, const
-                                                 cellsOnLayer.x[i] + delta_c,
-                                                 cellsOnLayer.y[i] - delta_c,
-                                                 cellsOnLayer.y[i] + delta_c);
-
-    for (int xBin = search_box[0]; xBin < search_box[1] + 1; ++xBin) {
-      for (int yBin = search_box[2]; yBin < search_box[3] + 1; ++yBin) {
-        int binId = lt.getGlobalBinByBin(xBin, yBin);
-        size_t binSize = lt[binId].size();
-
-        for (unsigned int j = 0; j < binSize; j++) {
-          unsigned int otherId = lt[binId][j];
-          if (distance(i, otherId, layerId) < delta_c) {
-            cellsOnLayer.rho[i] += (i == otherId ? 1.f : 0.5f) * cellsOnLayer.weight[otherId];
+    float delta(0);
+    if (cellsOnLayer.isSi[i]) {
+      delta = delta_c;
+      std::array<int, 4> search_box = lt.searchBox(
+          cellsOnLayer.x[i] - delta, cellsOnLayer.x[i] + delta, cellsOnLayer.y[i] - delta, cellsOnLayer.y[i] + delta);
+
+      for (int xBin = search_box[0]; xBin < search_box[1] + 1; ++xBin) {
+        for (int yBin = search_box[2]; yBin < search_box[3] + 1; ++yBin) {
+          int binId = lt.getGlobalBinByBin(xBin, yBin);
+          size_t binSize = lt[binId].size();
+
+          for (unsigned int j = 0; j < binSize; j++) {
+            unsigned int otherId = lt[binId][j];
+            if (!cellsOnLayer.isSi[otherId])
+              continue;  //cells in the silicon cannot talk to cells in scintillator
+            if (distance(i, otherId, layerId, false) < delta) {
+              cellsOnLayer.rho[i] += (i == otherId ? 1.f : 0.5f) * cellsOnLayer.weight[otherId];
+            }
+          }
+        }
+      }
+    } else {
+      delta = delta_r;
+      std::array<int, 4> search_box = lt.searchBoxEtaPhi(cellsOnLayer.eta[i] - delta,
+                                                         cellsOnLayer.eta[i] + delta,
+                                                         cellsOnLayer.phi[i] - delta,
+                                                         cellsOnLayer.phi[i] + delta);
+      cellsOnLayer.rho[i] += cellsOnLayer.weight[i];
+      float northeast(0), northwest(0), southeast(0), southwest(0), all(0);
+
+      for (int etaBin = search_box[0]; etaBin < search_box[1] + 1; ++etaBin) {
+        for (int phiBin = search_box[2]; phiBin < search_box[3] + 1; ++phiBin) {
+          int binId = lt.getGlobalBinByBinEtaPhi(etaBin, phiBin);
+          size_t binSize = lt[binId].size();
+
+          for (unsigned int j = 0; j < binSize; j++) {
+            unsigned int otherId = lt[binId][j];
+            if (cellsOnLayer.isSi[otherId])
+              continue;  //cells in the silicon cannot talk to cells in scintillator
+            if (distance(i, otherId, layerId, true) < delta) {
+              // need to round these numbers to 5 decimals to be able to compare floats
+              constexpr float roundDec = 100000.;
```

Comment:
this is not clear to me. What kind of property does a 5 digits number has, that a standard 7 digits float does not have?

Reply:
As explained below, the only reason I do rounding is to be able to compare two floats. I don't know 7 digits is the standard and I just thought round to 5 digits is safe. If 7 digits float is the standard, I just round it to 7 digits?